contracts/discovery/GNS.sol

```diff
@@ -359,175 +332,144 @@ contract GNS is GNSV1Storage, GraphUpgradeable, IGNS {
         uint256 _tokensOutMin
     ) external override notPartialPaused {
         // Pool checks
-        NameCurationPool storage namePool = nameSignals[_graphAccount][_subgraphNumber];
+        SubgraphData storage namePool = _getSubgraphPool(_subgraphID);
         require(namePool.disabled == false, "GNS: Cannot be disabled");
 
         // Curator balance checks
-        uint256 curatorNSignal = namePool.curatorNSignal[msg.sender];
+        address curator = msg.sender;
+        uint256 curatorNSignal = namePool.curatorNSignal[curator];
         require(
             _nSignal <= curatorNSignal,
             "GNS: Curator cannot withdraw more nSignal than they have"
         );
 
         // Get tokens for name signal amount to burn
-        uint256 vSignal = nSignalToVSignal(_graphAccount, _subgraphNumber, _nSignal);
+        uint256 vSignal = nSignalToVSignal(_subgraphID, _nSignal);
         uint256 tokens = curation().burn(namePool.subgraphDeploymentID, vSignal, _tokensOutMin);
 
         // Update pools
         namePool.vSignal = namePool.vSignal.sub(vSignal);
         namePool.nSignal = namePool.nSignal.sub(_nSignal);
-        namePool.curatorNSignal[msg.sender] = namePool.curatorNSignal[msg.sender].sub(_nSignal);
+        namePool.curatorNSignal[curator] = namePool.curatorNSignal[curator].sub(_nSignal);
 
         // Return the tokens to the nameCurator
-        require(
-            graphToken().transfer(msg.sender, tokens),
-            "GNS: Error sending nameCurators tokens"
-        );
+        require(graphToken().transfer(curator, tokens), "GNS: Error sending nameCurators tokens");
 
-        emit NSignalBurned(_graphAccount, _subgraphNumber, msg.sender, _nSignal, vSignal, tokens);
+        emit SignalBurned(_subgraphID, curator, _nSignal, vSignal, tokens);
     }
 
-    /**
-     * @dev Owner disables the subgraph. This means the subgraph-number combination can no longer
-     * be used for name signal. The nSignal curve is destroyed, the vSignal is burned, and the GNS
-     * contract holds the GRT from burning the vSignal, which all curators can withdraw manually.
-     * @param _graphAccount Account that is deprecating its name curation
-     * @param _subgraphNumber Subgraph number
-     */
-    function _disableNameSignal(address _graphAccount, uint256 _subgraphNumber) private {
-        NameCurationPool storage namePool = nameSignals[_graphAccount][_subgraphNumber];
-
-        // If no nSignal, then no need to burn vSignal
-        if (namePool.nSignal != 0) {
-            namePool.withdrawableGRT = curation().burn(
-                namePool.subgraphDeploymentID,
-                namePool.vSignal,
-                0
-            );
-            namePool.vSignal = 0;
-        }
-
-        // Set the NameCurationPool fields to make it disabled
-        namePool.disabled = true;
-
-        emit NameSignalDisabled(_graphAccount, _subgraphNumber, namePool.withdrawableGRT);
+    // TODO: review this function
```

Comment:
Is this in reference to the fact we will have to handle V1 subgraphs, as well as the NFT subgraphs?

Seems complex. But doable

Reply:
`_getSubgraphPool` will give you the storage for the v1 subgraphs or the v2 subgraphs depending when it was created. We can't migrate old curators to the new struct.